enzyme/Enzyme/EnzymeLogic.cpp

```diff
@@ -2506,11 +2504,11 @@ void createInvertedTerminator(TypeResults &TR, DiffeGradientUtils *gutils,
 Function *EnzymeLogic::CreatePrimalAndGradient(
     Function *todiff, DIFFE_TYPE retType,
     const std::vector<DIFFE_TYPE> &constant_args, TargetLibraryInfo &TLI,
-    TypeAnalysis &TA, bool returnUsed, bool dretPtr, bool topLevel,
+    TypeAnalysis &TA, bool returnUsed, bool dretPtr, DerivativeMode mode,
     llvm::Type *additionalArg, const FnTypeInfo &oldTypeInfo_,
     const std::map<Argument *, bool> _uncacheable_args,
-    const AugmentedReturn *augmenteddata, bool AtomicAdd, bool fwdMode,
-    bool PostOpt, bool omp) {
+    const AugmentedReturn *augmenteddata, bool AtomicAdd, bool PostOpt,
+    bool omp) {
 
```

Comment:
This should also have an assertion about being reversemodegradient reversemodecombined (and for now forwardmode).

Reply:
done